skrf/media/media.py

```diff
@@ -405,9 +405,11 @@ def match(self, nports: int = 1, z0: Union[NumberLike, None] = None,
         result.frequency = self.frequency
         result.s = npy.zeros((self.frequency.npoints, nports, nports), dtype=complex)
         if z0 is None:
+            # z0 = npy.zeros(result.s.shape[:2])
+            # z0[:] = self.z0[:, None]
             z0 = self.z0
```

Comment:
I wanted to, but the commented code failed lot of tests, I'll dig into this issue later.

Reply:
I think it could be left as is (because it is not in `Network`) and will be solved when refactoring `Media`.
`media.z0` can be `None` and so far I have also noticed depending the way `z0` of media is defined, it can raise an exception in `Network.renormalize(media.z0)` because of this line:
https://github.com/scikit-rf/scikit-rf/blob/ba3b0a7f391838dabaa6c122db1a8e28067e9494/skrf/network.py#L3036
In some conditions (with list instead numpy arrays) the result is a bool and has no method `any()`.

I do not know of course if it is the issue you encountered, but @Ttl suggested the line could be written like below so I mention it just in case.
```Python
if npy.any(self.z0 != z_new):
```

Reply:
Great PR anyway!